src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -134,6 +150,44 @@ public PrivateKey getCAPrivateKey() {
         return caPrivateKey;
     }
 
+
+
+    /**
+     * Get certificate chain using private and certificate URIs.
+     *
+     * @param privateKeyUri  private key URI
+     * @param certificateUri certificate URI
+     * @return X509Certificate list that corresponds to the given private key and certificate URIs
+     * @throws KeyLoadingException              if it fails to load key
+     * @throws CertificateChainLoadingException if no certificate chain is found
+     * @throws ServiceUnavailableException     if the security service is not available
+     */
+    public  X509Certificate[] loadCaCertificateChain(URI privateKeyUri, URI certificateUri)
+            throws CertificateChainLoadingException, KeyLoadingException, ServiceUnavailableException {
+        KeyManager[] km = securityService.getKeyManagers(privateKeyUri, certificateUri);
```

Comment:
Yes, good. Now that this is using the security service it will work correctly when the device is configured to use an HSM.